lib/spack/spack/directives.py

```diff
@@ -486,6 +486,9 @@ def _depends_on(
         return
 
     dep_spec = spack.spec.Spec(spec)
+
+    if dep_spec.name in ("c", "cxx", "fortran"):
```

Comment:
I think we just need:
```suggestion
    if dep_spec.name == "c":
```
Both `cxx` and `fortran` are already treated in a special way, that brings in some runtime libs consistency, see `SUPPORTED_LANGUAGES` in the same file.

Reply:
Oh. I think it's a bad idea Spec(...) is called twice :|